### PR TITLE
fix signal handling

### PR DESCRIPTION
**Pull request by lowne**

### CODE REVIEW

File: vdu_controls.py [hunk 1]
```diff
@@ -3265,7 +3265,7 @@ def respond_to_unix_signal(signal_number: int):
             if signal_number == signal.SIGHUP:
                 self.main_control_panel.refresh_data()
             elif PRESET_SIGNAL_MIN <= signal_number <= PRESET_SIGNAL_MAX:
-                restore_preset = self.preset_controller.get_preset_name(signal_number - PRESET_SIGNAL_MIN)
+                restore_preset = self.preset_controller.get_preset(signal_number - PRESET_SIGNAL_MIN)
```
Comment (digitaltrails) [inline vdu_controls.py:3268]: This seems to be the required fix.

File: vdu_controls.py [hunk 2]
```diff
@@ -2510,7 +2510,7 @@ def delete_preset(self, preset: Preset) -> None:
     def get_preset(self, preset_number: int) -> Preset | None:
         presets = self.find_presets()
         if preset_number < len(presets):
-            return presets.values()[preset_number]
+            return list(presets.values())[preset_number]
```
Comment (digitaltrails) [inline vdu_controls.py:2513]: Interesting.  Would not have thought this was necessary.  Will test.

File: vdu_controls.py [hunk 3]
```diff
@@ -560,21 +560,21 @@ def translate(source_text: str):
...
-settings or combinations of settings may cause physical effects including and not 
-limited to overheating, screen-burn-in, shortened backlight lifetime, and cease 
+Enabling ddcutil for VCP-codes not in the Display Data Channel (DDC) Virtual Control
+Panel (VCP) standard may result in irreversible damage to any connected monitors and
+any devices they are connected to (including the driving PC).  Before enabling
+non-standard codes one should consider that these codes may need to be operated
+in conjunction with other settings or codes. One should also be mindful that some
+settings or combinations of settings may cause physical effects including and not
+limited to overheating, screen-burn-in, shortened backlight lifetime, and cease
 of function (bricking).
```
Comment (digitaltrails) [inline vdu_controls.py:570]: Have to check this does not break enabling a semi-secret option.

**Comment by digitaltrails** [comment]
Thanks for the fix.  

I was a bit worried the stripping of trailing spaces might break a bit of special code, but all is well.  But it seems fine.  As you can see I've merged the changes.  

I will make an official release sometime today, then update the OpenSUSE build, and hopefully the AUR should also update within a day or so.

File: vdu_controls.py (hunk 2 above)
Comment (digitaltrails) [reply vdu_controls.py:2513]: Oh, yes, got bitten by this previously.

**Comment by lowne** [comment]
> I was a bit worried the stripping of trailing spaces might break a bit of special code,

Sorry for that, I was trying the fancy github integration in vscode and had assumed that some sort of preview would pop up when i clicked the PR button, but alas that's not a thing - or maybe I need to configure it somewhere.
OTOH, having *semantic* trailing whitespace seems like the most delicious evil :smiley:  

> the AUR should also update

It did :+1:

**Comment by digitaltrails** [comment]
> > I was a bit worried the stripping of trailing spaces might break a bit of special code,
> 
> Sorry for that, I was trying the fancy github integration in vscode and had assumed that some sort of preview would pop up when i clicked the PR button, but alas that's not a thing - or maybe I need to configure it somewhere. OTOH, having _semantic_ trailing whitespace seems like the most delicious evil smiley
> 

The trailing spaces was a mistake on my part and one of the constants is used in a test, but no one except me is likely using that code down that branch (for experimenting with codes that might not be safe).

Brings to mind the  the [Vodafone Greece Affair](https://spectrum.ieee.org/the-athens-affair), where someone changed the switch's command parser to turn off logging if a command was followed by six spaces - deadly evil in that instance.

> > the AUR should also update
> 
> It did +1

Mark Wagie is very quick on picking up new versions.

**Pull request merged**